eZ/Publish/API/Repository/Values/ContentType/ContentType.php

```diff
@@ -184,7 +184,49 @@ abstract class ContentType extends ValueObject implements MultiLanguageName, Mul
      *
      * @param string $fieldDefinitionIdentifier
      *
-     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition|null
      */
     abstract public function getFieldDefinition($fieldDefinitionIdentifier);
+
+    /**
+     * This method returns true if the field definition for the given identifier exists.
+     *
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return bool
+     */
+    public function hasFieldDefinition(string $fieldDefinitionIdentifier): bool
+    {
+        return $this->getFieldDefinition($fieldDefinitionIdentifier) !== null;
+    }
+
+    /**
+     * Returns first field definition with given field type identifier or null.
+     *
+     * @param string $fieldTypeIdentifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition
+     */
+    public function getFieldDefinitionWithType(string $fieldTypeIdentifier): ?FieldDefinition
+    {
+        foreach ($this->getFieldDefinitions() as $fieldDefinition) {
+            if ($fieldDefinition->fieldTypeIdentifier === $fieldTypeIdentifier) {
+                return $fieldDefinition;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns true if field definition with given field type exists.
+     *
+     * @param string $fieldTypeIdentifier
+     *
+     * @return bool
+     */
+    public function hasFieldDefinitionWithType(string $fieldTypeIdentifier): bool
```

Comment:
Why not returning the number of fields of the given type? The returned number can be used in the same IFs as a boolean would do. It also allows for more complex scenarios (eg. a Content has 2 images => which one to map for our needs ?)

Reply:
Also: given that this method does not optimize away any processing, as it is seems a bit pointless... The user might just put in her code the same check `if ($ft->getFieldDefinitionWithType($fieldTypeIdentifier) !== null)`

Reply:
It would sound good with a `countFieldDefinitionsWithType()`.